sdk/python/kfp/_client.py

```diff
@@ -255,8 +255,21 @@ def _load_context_setting_or_default(self):
         self._context_setting = json.load(f)
     else:
       self._context_setting = {
-        'namespace': '',
+        'namespace': self._guess_user_namespace_or_default(),
       }
+
+  def _guess_user_namespace_or_default(self):
+    """Guess user namespace from environment variable NB_PREFIX"""
```

Comment:
This is good. Have same feeling, setting namespace explicitly is kind of unnecessary.

This implementation uses business logic and has the dependency on how to build jupyter notebook. 
Do you think it's better to use an application neutral way? 

1. check file `/var/run/secrets/kubernetes.io/serviceaccount/namespace` 
2. Change notebook side and already add ENN POD_NAMESPACE and here, we can just read `POD_NAMESPACE`

Reply:
I absolutely agree about using an application neutral way.
Using the notebook virtualservice path seems brittle.

Reply:
+1 to this suggestion

Reply:
Good idea! addressed in https://github.com/kubeflow/pipelines/pull/4293/commits/8575a57fce9c4527e217f0eca32357141d3e5f77